assets/js/googlesitekit/datastore/user/surveys.js

```diff
@@ -0,0 +1,154 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import isPlainObject from 'lodash/isPlainObject';
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../data/create-fetch-store';
+import { createValidatedAction } from '../../data/utils';
+import { createCacheKey } from '../../api';
+import { getItem, setItem } from '../../api/cache';
+
+const fetchTriggerSurveyStore = createFetchStore( {
+	baseName: 'triggerSurvey',
+	controlCallback: ( { triggerID } ) => {
+		return API.set( 'core', 'user', 'survey-trigger', { triggerID } );
+	},
+	argsToParams: ( triggerID ) => {
+		return { triggerID };
+	},
+	reducerCallback: ( state, { survey_payload, session } ) => { // eslint-disable-line camelcase
+		return {
+			...state,
+			currentSurvey: survey_payload,
+			currentSurveySession: session,
+		};
+	},
+	validateParams: ( { triggerID } = {} ) => {
+		invariant( 'string' === typeof triggerID, 'triggerID is mandatory and must be a string' );
+	},
+
+} );
+
+const fetchSendSurveyEventStore = createFetchStore( {
+	baseName: 'sendSurveyEvent',
+	controlCallback: ( { event, session } ) => API.set( 'core', 'user', 'survey-event', { event, session } ),
+	argsToParams: ( event, session ) => {
+		return { event, session };
+	},
+} );
+
+const baseInitialState = {
+	currentSurvey: null,
+	currentSurveySession: null,
+};
+
+const baseActions = {
+	triggerSurvey: createValidatedAction(
+		( triggerID, options = { ttl: 0 } ) => {
+			invariant( 'string' === typeof triggerID, 'triggerID is mandatory and must be a string' );
+			invariant( isPlainObject( options ), 'options must be an object' );
+			invariant( 'number' === typeof options.ttl, 'options.ttl must be a number' );
+		},
+		function* ( triggerID, options = { ttl: 0 } ) {
+			const { select } = yield Data.commonActions.getRegistry();
+			if ( null !== select( STORE_NAME ).getCurrentSurvey() ) {
+				return;
+			}
+			const cacheKey = createCacheKey( 'core', 'user', 'survey-event', { triggerID } );
+			const { cacheHit, value } = yield Data.commonActions.await( getItem( cacheKey ) );
+			if ( false === cacheHit && options.ttl ) {
+				const { error, response } = yield fetchTriggerSurveyStore.actions.fetchTriggerSurvey( triggerID );
+				if ( ! error && options.ttl > 0 ) {
+					yield Data.commonActions.await( setItem( cacheKey, {} ) );
+					return { response, error };
+				}
+			}
+			return {
+				response: value,
```

Comment:
Not sure we should return `value` as the response here since value is from the cache which doesn't get the response 😄 

Maybe we should just return an empty object here (and in the early return above) as we do in some other actions.

Reply:
Good catch, https://github.com/google/site-kit-wp/pull/3431/commits/01de7a94811c89adf061c7e9c8de7ae43d70dc33